{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XUZS4GUkHIEG"
      },
      "source": [
        "# Подготовка файлов и программ\n",
        "\n",
        "Некоторые программы ставятся дважды из разных источников для демонстрации различных вариантов в случае неполадок. Для оптимизации времени работы необходимо удалить лишний код."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pV47sHC0DdT1",
        "outputId": "6a8ff003-b2f1-48c1-e044-19f469f868f7"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting macs2\n",
            "  Using cached MACS2-2.2.7.1.tar.gz (2.1 MB)\n",
            "  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  \u001b[1;31merror\u001b[0m: \u001b[1msubprocess-exited-with-error\u001b[0m\n",
            "  \n",
            "  \u001b[31m×\u001b[0m \u001b[32mPreparing metadata \u001b[0m\u001b[1;32m(\u001b[0m\u001b[32mpyproject.toml\u001b[0m\u001b[1;32m)\u001b[0m did not run successfully.\n",
            "  \u001b[31m│\u001b[0m exit code: \u001b[1;36m1\u001b[0m\n",
            "  \u001b[31m╰─>\u001b[0m See above for output.\n",
            "  \n",
            "  \u001b[1;35mnote\u001b[0m: This error originates from a subprocess, and is likely not a problem with pip.\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25herror\n",
            "\u001b[1;31merror\u001b[0m: \u001b[1mmetadata-generation-failed\u001b[0m\n",
            "\n",
            "\u001b[31m×\u001b[0m Encountered error while generating package metadata.\n",
            "\u001b[31m╰─>\u001b[0m See above for output.\n",
            "\n",
            "\u001b[1;35mnote\u001b[0m: This is an issue with the package mentioned above, not pip.\n",
            "\u001b[1;36mhint\u001b[0m: See above for details.\n"
          ]
        }
      ],
      "source": [
        "!pip install macs2"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lrtRbP3fAvj8",
        "outputId": "9169c893-b0cc-4de7-bb62-8efe634d38fb"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "!pip install -q condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "jQ1Oqzb9BXoG"
      },
      "outputs": [],
      "source": [
        "import condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q5FUOHCgBZWp",
        "outputId": "aeb111e0-94ea-4c3d-ce8a-327529489ab9"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "✨🍰✨ Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XnbIt77YBOT9",
        "outputId": "b45391b5-56e9-43a4-d14e-6b770c42bad9"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda seqtk"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sCBV_31qHOik",
        "outputId": "2368def4-6176-4b48-dd07-9721636c1212"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda trimmomatic"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F0BLWsH2n3Ov",
        "outputId": "31415512-5c8c-449c-922f-fa9eacc578ab"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Requirement already satisfied: intervene in /usr/local/lib/python3.8/site-packages (0.6.5)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.8/site-packages (from intervene) (1.5.3)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.8/site-packages (from intervene) (1.24.2)\n",
            "Requirement already satisfied: pybedtools in /usr/local/lib/python3.8/site-packages (from intervene) (0.9.0)\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.8/site-packages (from intervene) (3.7.1)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.8/site-packages (from intervene) (1.10.1)\n",
            "Requirement already satisfied: seaborn in /usr/local/lib/python3.8/site-packages (from intervene) (0.12.2)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (23.0)\n",
            "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (9.4.0)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (1.0.7)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (4.38.0)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (0.11.0)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (1.4.4)\n",
            "Requirement already satisfied: importlib-resources>=3.2.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (5.12.0)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (2.8.2)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (3.0.9)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/site-packages (from pandas->intervene) (2022.7.1)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.8/site-packages (from pybedtools->intervene) (1.16.0)\n",
            "Requirement already satisfied: pysam in /usr/local/lib/python3.8/site-packages (from pybedtools->intervene) (0.20.0)\n",
            "Requirement already satisfied: zipp>=3.1.0 in /usr/local/lib/python3.8/site-packages (from importlib-resources>=3.2.0->matplotlib->intervene) (3.15.0)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "!pip install intervene"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ce6hiSTdkarU",
        "outputId": "34033297-5e51-4cee-ff45-1e57861d8e88"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "bedtools is already the newest version (2.27.1+dfsg-4ubuntu1).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 22 not upgraded.\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AUML3dvsqT9E",
        "outputId": "c2c259da-9655-4d96-a4ac-f268c8f2d916"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  cwltool fonts-glyphicons-halflings libc-ares2 libhts3 libjs-bootstrap\n",
            "  libnode64 nodejs nodejs-doc python3-bs4 python3-cachecontrol\n",
            "  python3-coloredlogs python3-dateutil python3-decorator python3-html5lib\n",
            "  python3-humanfriendly python3-isodate python3-lockfile python3-lxml\n",
            "  python3-mistune python3-msgpack python3-mypy-extensions python3-networkx\n",
            "  python3-prov python3-psutil python3-pyparsing python3-rdflib\n",
            "  python3-rdflib-jsonld python3-ruamel.yaml python3-schema-salad\n",
            "  python3-shellescape python3-soupsieve python3-sparqlwrapper\n",
            "  python3-typing-extensions python3-webencodings python3-yaml\n",
            "Suggested packages:\n",
            "  bowtie2-examples docker.io | singularity-container npm python3-genshi\n",
            "  python-lockfile-doc python3-lxml-dbg python-lxml-doc python-networkx-doc\n",
            "  python3-matplotlib python3-pygraphviz | python3-pydot python3-scipy\n",
            "  python-prov-doc python3-pydot python-psutil-doc python-pyparsing-doc\n",
            "  python-rdflib-doc\n",
            "The following NEW packages will be installed:\n",
            "  bowtie2 cwltool fonts-glyphicons-halflings libc-ares2 libhts3\n",
            "  libjs-bootstrap libnode64 nodejs nodejs-doc python3-bs4 python3-cachecontrol\n",
            "  python3-coloredlogs python3-dateutil python3-decorator python3-html5lib\n",
            "  python3-humanfriendly python3-isodate python3-lockfile python3-lxml\n",
            "  python3-mistune python3-msgpack python3-mypy-extensions python3-networkx\n",
            "  python3-prov python3-psutil python3-pyparsing python3-rdflib\n",
            "  python3-rdflib-jsonld python3-ruamel.yaml python3-schema-salad\n",
            "  python3-shellescape python3-soupsieve python3-sparqlwrapper\n",
            "  python3-typing-extensions python3-webencodings python3-yaml samtools\n",
            "0 upgraded, 37 newly installed, 0 to remove and 22 not upgraded.\n",
            "Need to get 13.4 MB of archives.\n",
            "After this operation, 64.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 python3-yaml amd64 5.3.1-1ubuntu0.1 [136 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-humanfriendly all 4.18-2 [60.8 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-coloredlogs all 7.3-2 [28.0 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-mypy-extensions all 0.4.3-1 [5,436 B]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-dateutil all 2.7.3-3ubuntu1 [63.3 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 python3-lxml amd64 4.5.0-1ubuntu0.5 [1,384 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-decorator all 4.4.2-0ubuntu1 [10.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-networkx all 2.4-3ubuntu1 [855 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-isodate all 0.6.0-2 [24.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-pyparsing all 2.4.6-1 [61.3 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-rdflib all 4.2.2-5 [241 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-prov all 1.5.2-2 [92.8 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-psutil amd64 5.5.1-1ubuntu4 [141 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-ruamel.yaml amd64 0.15.89-3build1 [228 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-msgpack amd64 0.6.2-1 [73.5 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-lockfile all 1:0.12.2-2ubuntu2 [14.6 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-cachecontrol all 0.12.6-1ubuntu2 [16.7 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-mistune all 0.8.4-2 [14.3 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-rdflib-jsonld all 0.4.0-5 [12.4 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-typing-extensions all 3.7.4.1-1 [27.5 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu focal/universe amd64 fonts-glyphicons-halflings all 1.009~3.4.1+dfsg-1 [117 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu focal/universe amd64 libjs-bootstrap all 3.4.1+dfsg-1 [124 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-schema-salad all 5.0.20200220195218-1 [295 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-shellescape all 3.4.1-4 [4,928 B]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu focal/universe amd64 cwltool all 2.0.20200224214940+dfsg-1 [203 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 libhts3 amd64 1.10.2-3ubuntu0.1 [350 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libc-ares2 amd64 1.15.0-1ubuntu0.2 [36.7 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu focal/universe amd64 libnode64 amd64 10.19.0~dfsg-3ubuntu1 [5,765 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu focal/universe amd64 nodejs-doc all 10.19.0~dfsg-3ubuntu1 [942 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-soupsieve all 1.9.5+dfsg-1 [29.1 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-bs4 all 4.8.2-1 [83.0 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-webencodings all 0.5.1-1ubuntu1 [11.0 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-html5lib all 1.0.1-2 [84.3 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-sparqlwrapper all 1.8.5-1 [32.3 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu focal/universe amd64 samtools amd64 1.10-3 [496 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu focal/universe amd64 bowtie2 amd64 2.3.5.1-6build1 [1,324 kB]\n",
            "Get:37 http://archive.ubuntu.com/ubuntu focal/universe amd64 nodejs amd64 10.19.0~dfsg-3ubuntu1 [61.1 kB]\n",
            "Fetched 13.4 MB in 2s (7,424 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python3-yaml.\n",
            "(Reading database ... 128268 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python3-yaml_5.3.1-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking python3-yaml (5.3.1-1ubuntu0.1) ...\n",
            "Selecting previously unselected package python3-humanfriendly.\n",
            "Preparing to unpack .../01-python3-humanfriendly_4.18-2_all.deb ...\n",
            "Unpacking python3-humanfriendly (4.18-2) ...\n",
            "Selecting previously unselected package python3-coloredlogs.\n",
            "Preparing to unpack .../02-python3-coloredlogs_7.3-2_all.deb ...\n",
            "Unpacking python3-coloredlogs (7.3-2) ...\n",
            "Selecting previously unselected package python3-mypy-extensions.\n",
            "Preparing to unpack .../03-python3-mypy-extensions_0.4.3-1_all.deb ...\n",
            "Unpacking python3-mypy-extensions (0.4.3-1) ...\n",
            "Selecting previously unselected package python3-dateutil.\n",
            "Preparing to unpack .../04-python3-dateutil_2.7.3-3ubuntu1_all.deb ...\n",
            "Unpacking python3-dateutil (2.7.3-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-lxml:amd64.\n",
            "Preparing to unpack .../05-python3-lxml_4.5.0-1ubuntu0.5_amd64.deb ...\n",
            "Unpacking python3-lxml:amd64 (4.5.0-1ubuntu0.5) ...\n",
            "Selecting previously unselected package python3-decorator.\n",
            "Preparing to unpack .../06-python3-decorator_4.4.2-0ubuntu1_all.deb ...\n",
            "Unpacking python3-decorator (4.4.2-0ubuntu1) ...\n",
            "Selecting previously unselected package python3-networkx.\n",
            "Preparing to unpack .../07-python3-networkx_2.4-3ubuntu1_all.deb ...\n",
            "Unpacking python3-networkx (2.4-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-isodate.\n",
            "Preparing to unpack .../08-python3-isodate_0.6.0-2_all.deb ...\n",
            "Unpacking python3-isodate (0.6.0-2) ...\n",
            "Selecting previously unselected package python3-pyparsing.\n",
            "Preparing to unpack .../09-python3-pyparsing_2.4.6-1_all.deb ...\n",
            "Unpacking python3-pyparsing (2.4.6-1) ...\n",
            "Selecting previously unselected package python3-rdflib.\n",
            "Preparing to unpack .../10-python3-rdflib_4.2.2-5_all.deb ...\n",
            "Unpacking python3-rdflib (4.2.2-5) ...\n",
            "Selecting previously unselected package python3-prov.\n",
            "Preparing to unpack .../11-python3-prov_1.5.2-2_all.deb ...\n",
            "Unpacking python3-prov (1.5.2-2) ...\n",
            "Selecting previously unselected package python3-psutil.\n",
            "Preparing to unpack .../12-python3-psutil_5.5.1-1ubuntu4_amd64.deb ...\n",
            "Unpacking python3-psutil (5.5.1-1ubuntu4) ...\n",
            "Selecting previously unselected package python3-ruamel.yaml.\n",
            "Preparing to unpack .../13-python3-ruamel.yaml_0.15.89-3build1_amd64.deb ...\n",
            "Unpacking python3-ruamel.yaml (0.15.89-3build1) ...\n",
            "Selecting previously unselected package python3-msgpack.\n",
            "Preparing to unpack .../14-python3-msgpack_0.6.2-1_amd64.deb ...\n",
            "Unpacking python3-msgpack (0.6.2-1) ...\n",
            "Selecting previously unselected package python3-lockfile.\n",
            "Preparing to unpack .../15-python3-lockfile_1%3a0.12.2-2ubuntu2_all.deb ...\n",
            "Unpacking python3-lockfile (1:0.12.2-2ubuntu2) ...\n",
            "Selecting previously unselected package python3-cachecontrol.\n",
            "Preparing to unpack .../16-python3-cachecontrol_0.12.6-1ubuntu2_all.deb ...\n",
            "Unpacking python3-cachecontrol (0.12.6-1ubuntu2) ...\n",
            "Selecting previously unselected package python3-mistune.\n",
            "Preparing to unpack .../17-python3-mistune_0.8.4-2_all.deb ...\n",
            "Unpacking python3-mistune (0.8.4-2) ...\n",
            "Selecting previously unselected package python3-rdflib-jsonld.\n",
            "Preparing to unpack .../18-python3-rdflib-jsonld_0.4.0-5_all.deb ...\n",
            "Unpacking python3-rdflib-jsonld (0.4.0-5) ...\n",
            "Selecting previously unselected package python3-typing-extensions.\n",
            "Preparing to unpack .../19-python3-typing-extensions_3.7.4.1-1_all.deb ...\n",
            "Unpacking python3-typing-extensions (3.7.4.1-1) ...\n",
            "Selecting previously unselected package fonts-glyphicons-halflings.\n",
            "Preparing to unpack .../20-fonts-glyphicons-halflings_1.009~3.4.1+dfsg-1_all.deb ...\n",
            "Unpacking fonts-glyphicons-halflings (1.009~3.4.1+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../21-libjs-bootstrap_3.4.1+dfsg-1_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.4.1+dfsg-1) ...\n",
            "Selecting previously unselected package python3-schema-salad.\n",
            "Preparing to unpack .../22-python3-schema-salad_5.0.20200220195218-1_all.deb ...\n",
            "Unpacking python3-schema-salad (5.0.20200220195218-1) ...\n",
            "Selecting previously unselected package python3-shellescape.\n",
            "Preparing to unpack .../23-python3-shellescape_3.4.1-4_all.deb ...\n",
            "Unpacking python3-shellescape (3.4.1-4) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../24-cwltool_2.0.20200224214940+dfsg-1_all.deb ...\n",
            "Unpacking cwltool (2.0.20200224214940+dfsg-1) ...\n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "Preparing to unpack .../25-libhts3_1.10.2-3ubuntu0.1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.10.2-3ubuntu0.1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../26-libc-ares2_1.15.0-1ubuntu0.2_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.15.0-1ubuntu0.2) ...\n",
            "Selecting previously unselected package libnode64:amd64.\n",
            "Preparing to unpack .../27-libnode64_10.19.0~dfsg-3ubuntu1_amd64.deb ...\n",
            "Unpacking libnode64:amd64 (10.19.0~dfsg-3ubuntu1) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../28-nodejs-doc_10.19.0~dfsg-3ubuntu1_all.deb ...\n",
            "Unpacking nodejs-doc (10.19.0~dfsg-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-soupsieve.\n",
            "Preparing to unpack .../29-python3-soupsieve_1.9.5+dfsg-1_all.deb ...\n",
            "Unpacking python3-soupsieve (1.9.5+dfsg-1) ...\n",
            "Selecting previously unselected package python3-bs4.\n",
            "Preparing to unpack .../30-python3-bs4_4.8.2-1_all.deb ...\n",
            "Unpacking python3-bs4 (4.8.2-1) ...\n",
            "Selecting previously unselected package python3-webencodings.\n",
            "Preparing to unpack .../31-python3-webencodings_0.5.1-1ubuntu1_all.deb ...\n",
            "Unpacking python3-webencodings (0.5.1-1ubuntu1) ...\n",
            "Selecting previously unselected package python3-html5lib.\n",
            "Preparing to unpack .../32-python3-html5lib_1.0.1-2_all.deb ...\n",
            "Unpacking python3-html5lib (1.0.1-2) ...\n",
            "Selecting previously unselected package python3-sparqlwrapper.\n",
            "Preparing to unpack .../33-python3-sparqlwrapper_1.8.5-1_all.deb ...\n",
            "Unpacking python3-sparqlwrapper (1.8.5-1) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../34-samtools_1.10-3_amd64.deb ...\n",
            "Unpacking samtools (1.10-3) ...\n",
            "Selecting previously unselected package bowtie2.\n",
            "Preparing to unpack .../35-bowtie2_2.3.5.1-6build1_amd64.deb ...\n",
            "Unpacking bowtie2 (2.3.5.1-6build1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../36-nodejs_10.19.0~dfsg-3ubuntu1_amd64.deb ...\n",
            "Unpacking nodejs (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-lockfile (1:0.12.2-2ubuntu2) ...\n",
            "Setting up libhts3:amd64 (1.10.2-3ubuntu0.1) ...\n",
            "Setting up fonts-glyphicons-halflings (1.009~3.4.1+dfsg-1) ...\n",
            "Setting up python3-mypy-extensions (0.4.3-1) ...\n",
            "Setting up python3-yaml (5.3.1-1ubuntu0.1) ...\n",
            "Setting up libc-ares2:amd64 (1.15.0-1ubuntu0.2) ...\n",
            "Setting up python3-webencodings (0.5.1-1ubuntu1) ...\n",
            "Setting up python3-psutil (5.5.1-1ubuntu4) ...\n",
            "Setting up samtools (1.10-3) ...\n",
            "Setting up bowtie2 (2.3.5.1-6build1) ...\n",
            "Setting up python3-isodate (0.6.0-2) ...\n",
            "Setting up python3-decorator (4.4.2-0ubuntu1) ...\n",
            "Setting up python3-pyparsing (2.4.6-1) ...\n",
            "Setting up python3-humanfriendly (4.18-2) ...\n",
            "Setting up libnode64:amd64 (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-typing-extensions (3.7.4.1-1) ...\n",
            "Setting up python3-html5lib (1.0.1-2) ...\n",
            "Setting up nodejs-doc (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-ruamel.yaml (0.15.89-3build1) ...\n",
            "Setting up python3-lxml:amd64 (4.5.0-1ubuntu0.5) ...\n",
            "Setting up python3-rdflib (4.2.2-5) ...\n",
            "/usr/lib/python3/dist-packages/rdflib/plugins/memory.py:354: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  if triple is None or triple is (None,None,None):\n",
            "Setting up python3-dateutil (2.7.3-3ubuntu1) ...\n",
            "Setting up python3-msgpack (0.6.2-1) ...\n",
            "Setting up python3-cachecontrol (0.12.6-1ubuntu2) ...\n",
            "Setting up python3-shellescape (3.4.1-4) ...\n",
            "Setting up python3-coloredlogs (7.3-2) ...\n",
            "Setting up python3-mistune (0.8.4-2) ...\n",
            "Setting up python3-soupsieve (1.9.5+dfsg-1) ...\n",
            "Setting up libjs-bootstrap (3.4.1+dfsg-1) ...\n",
            "Setting up python3-rdflib-jsonld (0.4.0-5) ...\n",
            "Setting up python3-sparqlwrapper (1.8.5-1) ...\n",
            "Setting up python3-schema-salad (5.0.20200220195218-1) ...\n",
            "Setting up nodejs (10.19.0~dfsg-3ubuntu1) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python3-networkx (2.4-3ubuntu1) ...\n",
            "/usr/lib/python3/dist-packages/networkx/readwrite/tests/test_graphml.py:933: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  assert H.nodes['n1']['special'] is 0\n",
            "/usr/lib/python3/dist-packages/networkx/readwrite/tests/test_graphml.py:935: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  assert H.edges['n0','n1',1]['special'] is 0\n",
            "Setting up python3-bs4 (4.8.2-1) ...\n",
            "Setting up python3-prov (1.5.2-2) ...\n",
            "Setting up cwltool (2.0.20200224214940+dfsg-1) ...\n",
            "/usr/lib/python3/dist-packages/cwltool/docker.py:423: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  if res_req is not None and (\"ramMin\" in res_req or \"ramMax\" is res_req):\n",
            "Processing triggers for fontconfig (2.13.1-2ubuntu3) ...\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n",
            "Processing triggers for man-db (2.9.1-1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bowtie2 samtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JgTq4lFjq93N",
        "outputId": "0224b74e-a002-43c2-aa0c-9026dd76a299"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bowtie2\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bowtie2-2.5.1              |   py38h77f66f0_0        14.8 MB  bioconda\n",
            "    libhwloc-2.9.0             |       hd6dc26d_0         2.4 MB  conda-forge\n",
            "    perl-5.32.1                | 2_h7f98852_perl5        14.4 MB  conda-forge\n",
            "    tbb-2021.8.0               |       hf52228f_0         1.4 MB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        33.1 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.5.1-py38h77f66f0_0 \n",
            "  libhwloc           conda-forge/linux-64::libhwloc-2.9.0-hd6dc26d_0 \n",
            "  perl               conda-forge/linux-64::perl-5.32.1-2_h7f98852_perl5 \n",
            "  tbb                conda-forge/linux-64::tbb-2021.8.0-hf52228f_0 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "tbb-2021.8.0         | 1.4 MB    | :   0% 0/1 [00:00<?, ?it/s]\n",
            "bowtie2-2.5.1        | 14.8 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "\n",
            "libhwloc-2.9.0       | 2.4 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "tbb-2021.8.0         | 1.4 MB    | :   1% 0.010792658304727915/1 [00:00<00:24, 24.47s/it]\n",
            "bowtie2-2.5.1        | 14.8 MB   | :   0% 0.0010587414082268809/1 [00:00<04:13, 254.11s/it]\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :   0% 0.0010852662651499435/1 [00:00<04:05, 245.28s/it]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "libhwloc-2.9.0       | 2.4 MB    | :  72% 0.7150027177777735/1 [00:00<00:00,  2.58it/s]   \u001b[A\u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  14% 0.1386951244777214/1 [00:00<00:01,  2.08s/it]    \u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :  16% 0.1627899397724915/1 [00:00<00:01,  1.76s/it]    \u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  31% 0.3059762669775686/1 [00:00<00:00,  1.14s/it]\u001b[A\n",
            "\n",
            "\n",
            "tbb-2021.8.0         | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.25it/s]\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  50% 0.5039609103159953/1 [00:00<00:00,  1.23it/s]\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :  64% 0.6424776289687666/1 [00:00<00:00,  1.56it/s] \u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  73% 0.7305315716765478/1 [00:00<00:00,  1.56it/s]\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :  91% 0.9072825976653528/1 [00:00<00:00,  1.89it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "libhwloc-2.9.0       | 2.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.58it/s]               \u001b[A\u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | : 100% 1.0/1 [00:03<00:00,  5.33s/it]               \u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | : 100% 1.0/1 [00:03<00:00,  5.33s/it]\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\n",
            "Preparing transaction: - \b\b\\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda bowtie2"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vcr5CjibHTgF"
      },
      "source": [
        "## Установка FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8bpgwFgcbgRd",
        "outputId": "e17c7ef6-be1c-4ca4-db83-b05a1dc769c4"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2023-03-06 10:21:21--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  4.16MB/s    in 2.3s    \n",
            "\n",
            "2023-03-06 10:21:24 (4.16 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yTcTjOTJblcm",
        "outputId": "0d727f8a-08c8-40f0-a8d2-d100eca0fd17"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!unzip fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "S79LH0aXbvLx"
      },
      "outputs": [],
      "source": [
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Z56bxRHn5GXi",
        "outputId": "ade01fba-0783-46f8-dd2e-ac9611925ce6"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ],
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "10Fs3qrmHmJi"
      },
      "source": [
        "# Выравнивание"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-z4GQiMfU1wf"
      },
      "source": [
        "## Скачиваем чтения"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lFWnMg0uU0sJ",
        "outputId": "ae6ac030-4fac-43ff-903b-1e7c20e868f1"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2023-03-06 12:28:26--  https://www.encodeproject.org/files/ENCFF090UYV/@@download/ENCFF090UYV.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/10/20/1608ea35-7c3f-4a60-9127-8fb3b13391f3/ENCFF090UYV.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF090UYV.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4255YSP7&Signature=ymsgwJqNala4BA9%2BR%2BGicbH4nRM%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFwaCXVzLXdlc3QtMiJGMEQCIFP35A%2BioOrW6IVGyQyHYyYnK8VaWSvGp7jUchETbjZoAiBiZMoW%2BFvH5J%2FIfFXDLZbTnVIwEF8iC%2BQCVFHmBTmu2CqyBQgVEAAaDDIyMDc0ODcxNDg2MyIMpkRbfQzFqfchY2RgKo8F0aK9MbO1M4n%2BBobVCDvcscKUl3f5Wa28elm3mwEKjduocXa%2FNTbNLlzZVzvpGU50cW1buIQ%2BltwRhOlXU0izF%2FeUsCqIsXPNH4J4o8lTeUBG%2F%2FeFKlNnBXC0eqtqPUOiW%2FOBG66AvMFJRV590qz2RHCzOkBhSUcMnMNyZr3oqKl3Hkez%2Fq3R8tJ69K3EiZ4mPj7x81yrXgUUC3k6lAi%2B8q2t3BGylTFSLBJnLCEiz%2BVWy13EjkLOeh75y2HOtV6jLKpWJsqy3kKfJBUcf9ZYEFFElkr0%2BGDgmvYdQNp6U41VC2tGcV4%2BND1qrxVn5o50VArBZjgjSHsROXPZLlov5M%2FS5akXggIsxkTkzDb0qbLkzxFRyHwr4reiOin5vHsQTElkKGupSy7ML7qMZ1NiPxE3eIFa74hj8H2%2B4XMQcjfqrmnPCX98FdForKNZe8GCxowodixuSztEciKO8bCzpasUuSSuGIUNBnFb9ZoSLZvfQZ2Vp9eeXam7%2FxP3mLmpVUPnDj26oUW0v1JYeFNtvfJXFOtqBGe0Os9gwfjHlvZoJUMyIvYs95IllG7HyiW%2FheIQRshad1ppLkKv9ZsRR2LP2TGOuLramQ1PWd9nYc2K1Gd%2BZx%2BslQpCCUjWWuCb3meCpPbEk%2B7wDUFKh30hGqH3p4zuBR95zoGBUs3ol4WB5Qrp7XlemMcP4nfy4Aaz0%2BCg10nJu9VAk7nt33GNHFsOb2eRTaHS%2F%2BEkXgfeRNOgOBFhc6Hx44Fq1WAJQfcME8Oluug1gUJwPKvQJr8kTZneH9Bw8iCgFKbKWMYigrAikG1t869Qg%2FaVQrKXSkYW03DOIIhWUIl%2By8zsIcql6ivYxwRyGk%2FQWOJrlfMWZjD%2Bp5egBjqyAUYFoGFBCbsX9c7nEJ5z9ZCPWxlzRKSHcfSG1CiEobQmJTFtiCruv%2FKx%2BeiMKnFH0%2BiPEogMxL3Xf8RHgjWM%2F3oWUsl1Xmw%2FJpVGaNAP%2BkjXGNNqBKh%2FdavDTzgTyUZwIS8k2VgbXI6bITamkytADvU090sJnQ8wsR6VZ9QXGpbrBni8yZdWJFVeuBYURwiwZFKmv0ItFGd0dg4CLVaw2OZLFM8qv6EINLs9c6AohDjC%2FFY%3D&Expires=1678235306 [following]\n",
            "--2023-03-06 12:28:26--  https://encode-public.s3.amazonaws.com/2016/10/20/1608ea35-7c3f-4a60-9127-8fb3b13391f3/ENCFF090UYV.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF090UYV.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4255YSP7&Signature=ymsgwJqNala4BA9%2BR%2BGicbH4nRM%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFwaCXVzLXdlc3QtMiJGMEQCIFP35A%2BioOrW6IVGyQyHYyYnK8VaWSvGp7jUchETbjZoAiBiZMoW%2BFvH5J%2FIfFXDLZbTnVIwEF8iC%2BQCVFHmBTmu2CqyBQgVEAAaDDIyMDc0ODcxNDg2MyIMpkRbfQzFqfchY2RgKo8F0aK9MbO1M4n%2BBobVCDvcscKUl3f5Wa28elm3mwEKjduocXa%2FNTbNLlzZVzvpGU50cW1buIQ%2BltwRhOlXU0izF%2FeUsCqIsXPNH4J4o8lTeUBG%2F%2FeFKlNnBXC0eqtqPUOiW%2FOBG66AvMFJRV590qz2RHCzOkBhSUcMnMNyZr3oqKl3Hkez%2Fq3R8tJ69K3EiZ4mPj7x81yrXgUUC3k6lAi%2B8q2t3BGylTFSLBJnLCEiz%2BVWy13EjkLOeh75y2HOtV6jLKpWJsqy3kKfJBUcf9ZYEFFElkr0%2BGDgmvYdQNp6U41VC2tGcV4%2BND1qrxVn5o50VArBZjgjSHsROXPZLlov5M%2FS5akXggIsxkTkzDb0qbLkzxFRyHwr4reiOin5vHsQTElkKGupSy7ML7qMZ1NiPxE3eIFa74hj8H2%2B4XMQcjfqrmnPCX98FdForKNZe8GCxowodixuSztEciKO8bCzpasUuSSuGIUNBnFb9ZoSLZvfQZ2Vp9eeXam7%2FxP3mLmpVUPnDj26oUW0v1JYeFNtvfJXFOtqBGe0Os9gwfjHlvZoJUMyIvYs95IllG7HyiW%2FheIQRshad1ppLkKv9ZsRR2LP2TGOuLramQ1PWd9nYc2K1Gd%2BZx%2BslQpCCUjWWuCb3meCpPbEk%2B7wDUFKh30hGqH3p4zuBR95zoGBUs3ol4WB5Qrp7XlemMcP4nfy4Aaz0%2BCg10nJu9VAk7nt33GNHFsOb2eRTaHS%2F%2BEkXgfeRNOgOBFhc6Hx44Fq1WAJQfcME8Oluug1gUJwPKvQJr8kTZneH9Bw8iCgFKbKWMYigrAikG1t869Qg%2FaVQrKXSkYW03DOIIhWUIl%2By8zsIcql6ivYxwRyGk%2FQWOJrlfMWZjD%2Bp5egBjqyAUYFoGFBCbsX9c7nEJ5z9ZCPWxlzRKSHcfSG1CiEobQmJTFtiCruv%2FKx%2BeiMKnFH0%2BiPEogMxL3Xf8RHgjWM%2F3oWUsl1Xmw%2FJpVGaNAP%2BkjXGNNqBKh%2FdavDTzgTyUZwIS8k2VgbXI6bITamkytADvU090sJnQ8wsR6VZ9QXGpbrBni8yZdWJFVeuBYURwiwZFKmv0ItFGd0dg4CLVaw2OZLFM8qv6EINLs9c6AohDjC%2FFY%3D&Expires=1678235306\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.229.227, 52.218.219.3, 52.92.196.89, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.229.227|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1731397113 (1.6G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF090UYV.fastq.gz’\n",
            "\n",
            "ENCFF090UYV.fastq.g 100%[===================>]   1.61G  35.7MB/s    in 41s     \n",
            "\n",
            "2023-03-06 12:29:07 (40.5 MB/s) - ‘ENCFF090UYV.fastq.gz’ saved [1731397113/1731397113]\n",
            "\n",
            "--2023-03-06 12:29:08--  https://www.encodeproject.org/files/ENCFF815LPM/@@download/ENCFF815LPM.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/10/22/a50eb4e1-332c-4958-863e-102e1fb1e134/ENCFF815LPM.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF815LPM.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4255YSP7&Signature=I%2Fi1oYF%2BOBJi%2BHLISNK442Cxl10%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFwaCXVzLXdlc3QtMiJGMEQCIFP35A%2BioOrW6IVGyQyHYyYnK8VaWSvGp7jUchETbjZoAiBiZMoW%2BFvH5J%2FIfFXDLZbTnVIwEF8iC%2BQCVFHmBTmu2CqyBQgVEAAaDDIyMDc0ODcxNDg2MyIMpkRbfQzFqfchY2RgKo8F0aK9MbO1M4n%2BBobVCDvcscKUl3f5Wa28elm3mwEKjduocXa%2FNTbNLlzZVzvpGU50cW1buIQ%2BltwRhOlXU0izF%2FeUsCqIsXPNH4J4o8lTeUBG%2F%2FeFKlNnBXC0eqtqPUOiW%2FOBG66AvMFJRV590qz2RHCzOkBhSUcMnMNyZr3oqKl3Hkez%2Fq3R8tJ69K3EiZ4mPj7x81yrXgUUC3k6lAi%2B8q2t3BGylTFSLBJnLCEiz%2BVWy13EjkLOeh75y2HOtV6jLKpWJsqy3kKfJBUcf9ZYEFFElkr0%2BGDgmvYdQNp6U41VC2tGcV4%2BND1qrxVn5o50VArBZjgjSHsROXPZLlov5M%2FS5akXggIsxkTkzDb0qbLkzxFRyHwr4reiOin5vHsQTElkKGupSy7ML7qMZ1NiPxE3eIFa74hj8H2%2B4XMQcjfqrmnPCX98FdForKNZe8GCxowodixuSztEciKO8bCzpasUuSSuGIUNBnFb9ZoSLZvfQZ2Vp9eeXam7%2FxP3mLmpVUPnDj26oUW0v1JYeFNtvfJXFOtqBGe0Os9gwfjHlvZoJUMyIvYs95IllG7HyiW%2FheIQRshad1ppLkKv9ZsRR2LP2TGOuLramQ1PWd9nYc2K1Gd%2BZx%2BslQpCCUjWWuCb3meCpPbEk%2B7wDUFKh30hGqH3p4zuBR95zoGBUs3ol4WB5Qrp7XlemMcP4nfy4Aaz0%2BCg10nJu9VAk7nt33GNHFsOb2eRTaHS%2F%2BEkXgfeRNOgOBFhc6Hx44Fq1WAJQfcME8Oluug1gUJwPKvQJr8kTZneH9Bw8iCgFKbKWMYigrAikG1t869Qg%2FaVQrKXSkYW03DOIIhWUIl%2By8zsIcql6ivYxwRyGk%2FQWOJrlfMWZjD%2Bp5egBjqyAUYFoGFBCbsX9c7nEJ5z9ZCPWxlzRKSHcfSG1CiEobQmJTFtiCruv%2FKx%2BeiMKnFH0%2BiPEogMxL3Xf8RHgjWM%2F3oWUsl1Xmw%2FJpVGaNAP%2BkjXGNNqBKh%2FdavDTzgTyUZwIS8k2VgbXI6bITamkytADvU090sJnQ8wsR6VZ9QXGpbrBni8yZdWJFVeuBYURwiwZFKmv0ItFGd0dg4CLVaw2OZLFM8qv6EINLs9c6AohDjC%2FFY%3D&Expires=1678235348 [following]\n",
            "--2023-03-06 12:29:08--  https://encode-public.s3.amazonaws.com/2016/10/22/a50eb4e1-332c-4958-863e-102e1fb1e134/ENCFF815LPM.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF815LPM.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4255YSP7&Signature=I%2Fi1oYF%2BOBJi%2BHLISNK442Cxl10%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFwaCXVzLXdlc3QtMiJGMEQCIFP35A%2BioOrW6IVGyQyHYyYnK8VaWSvGp7jUchETbjZoAiBiZMoW%2BFvH5J%2FIfFXDLZbTnVIwEF8iC%2BQCVFHmBTmu2CqyBQgVEAAaDDIyMDc0ODcxNDg2MyIMpkRbfQzFqfchY2RgKo8F0aK9MbO1M4n%2BBobVCDvcscKUl3f5Wa28elm3mwEKjduocXa%2FNTbNLlzZVzvpGU50cW1buIQ%2BltwRhOlXU0izF%2FeUsCqIsXPNH4J4o8lTeUBG%2F%2FeFKlNnBXC0eqtqPUOiW%2FOBG66AvMFJRV590qz2RHCzOkBhSUcMnMNyZr3oqKl3Hkez%2Fq3R8tJ69K3EiZ4mPj7x81yrXgUUC3k6lAi%2B8q2t3BGylTFSLBJnLCEiz%2BVWy13EjkLOeh75y2HOtV6jLKpWJsqy3kKfJBUcf9ZYEFFElkr0%2BGDgmvYdQNp6U41VC2tGcV4%2BND1qrxVn5o50VArBZjgjSHsROXPZLlov5M%2FS5akXggIsxkTkzDb0qbLkzxFRyHwr4reiOin5vHsQTElkKGupSy7ML7qMZ1NiPxE3eIFa74hj8H2%2B4XMQcjfqrmnPCX98FdForKNZe8GCxowodixuSztEciKO8bCzpasUuSSuGIUNBnFb9ZoSLZvfQZ2Vp9eeXam7%2FxP3mLmpVUPnDj26oUW0v1JYeFNtvfJXFOtqBGe0Os9gwfjHlvZoJUMyIvYs95IllG7HyiW%2FheIQRshad1ppLkKv9ZsRR2LP2TGOuLramQ1PWd9nYc2K1Gd%2BZx%2BslQpCCUjWWuCb3meCpPbEk%2B7wDUFKh30hGqH3p4zuBR95zoGBUs3ol4WB5Qrp7XlemMcP4nfy4Aaz0%2BCg10nJu9VAk7nt33GNHFsOb2eRTaHS%2F%2BEkXgfeRNOgOBFhc6Hx44Fq1WAJQfcME8Oluug1gUJwPKvQJr8kTZneH9Bw8iCgFKbKWMYigrAikG1t869Qg%2FaVQrKXSkYW03DOIIhWUIl%2By8zsIcql6ivYxwRyGk%2FQWOJrlfMWZjD%2Bp5egBjqyAUYFoGFBCbsX9c7nEJ5z9ZCPWxlzRKSHcfSG1CiEobQmJTFtiCruv%2FKx%2BeiMKnFH0%2BiPEogMxL3Xf8RHgjWM%2F3oWUsl1Xmw%2FJpVGaNAP%2BkjXGNNqBKh%2FdavDTzgTyUZwIS8k2VgbXI6bITamkytADvU090sJnQ8wsR6VZ9QXGpbrBni8yZdWJFVeuBYURwiwZFKmv0ItFGd0dg4CLVaw2OZLFM8qv6EINLs9c6AohDjC%2FFY%3D&Expires=1678235348\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.178.201, 52.92.196.41, 52.92.177.209, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.178.201|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1428443569 (1.3G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF815LPM.fastq.gz’\n",
            "\n",
            "ENCFF815LPM.fastq.g 100%[===================>]   1.33G  42.5MB/s    in 34s     \n",
            "\n",
            "2023-03-06 12:29:42 (40.4 MB/s) - ‘ENCFF815LPM.fastq.gz’ saved [1428443569/1428443569]\n",
            "\n",
            "--2023-03-06 12:29:42--  https://www.encodeproject.org/files/ENCFF693RXC/@@download/ENCFF693RXC.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/10/21/c3715b23-624f-4b7e-b24d-64b0ef05ae7c/ENCFF693RXC.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF693RXC.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4255YSP7&Signature=ETkwLdnoBICtMvjrEOyWuv8TTQ8%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFwaCXVzLXdlc3QtMiJGMEQCIFP35A%2BioOrW6IVGyQyHYyYnK8VaWSvGp7jUchETbjZoAiBiZMoW%2BFvH5J%2FIfFXDLZbTnVIwEF8iC%2BQCVFHmBTmu2CqyBQgVEAAaDDIyMDc0ODcxNDg2MyIMpkRbfQzFqfchY2RgKo8F0aK9MbO1M4n%2BBobVCDvcscKUl3f5Wa28elm3mwEKjduocXa%2FNTbNLlzZVzvpGU50cW1buIQ%2BltwRhOlXU0izF%2FeUsCqIsXPNH4J4o8lTeUBG%2F%2FeFKlNnBXC0eqtqPUOiW%2FOBG66AvMFJRV590qz2RHCzOkBhSUcMnMNyZr3oqKl3Hkez%2Fq3R8tJ69K3EiZ4mPj7x81yrXgUUC3k6lAi%2B8q2t3BGylTFSLBJnLCEiz%2BVWy13EjkLOeh75y2HOtV6jLKpWJsqy3kKfJBUcf9ZYEFFElkr0%2BGDgmvYdQNp6U41VC2tGcV4%2BND1qrxVn5o50VArBZjgjSHsROXPZLlov5M%2FS5akXggIsxkTkzDb0qbLkzxFRyHwr4reiOin5vHsQTElkKGupSy7ML7qMZ1NiPxE3eIFa74hj8H2%2B4XMQcjfqrmnPCX98FdForKNZe8GCxowodixuSztEciKO8bCzpasUuSSuGIUNBnFb9ZoSLZvfQZ2Vp9eeXam7%2FxP3mLmpVUPnDj26oUW0v1JYeFNtvfJXFOtqBGe0Os9gwfjHlvZoJUMyIvYs95IllG7HyiW%2FheIQRshad1ppLkKv9ZsRR2LP2TGOuLramQ1PWd9nYc2K1Gd%2BZx%2BslQpCCUjWWuCb3meCpPbEk%2B7wDUFKh30hGqH3p4zuBR95zoGBUs3ol4WB5Qrp7XlemMcP4nfy4Aaz0%2BCg10nJu9VAk7nt33GNHFsOb2eRTaHS%2F%2BEkXgfeRNOgOBFhc6Hx44Fq1WAJQfcME8Oluug1gUJwPKvQJr8kTZneH9Bw8iCgFKbKWMYigrAikG1t869Qg%2FaVQrKXSkYW03DOIIhWUIl%2By8zsIcql6ivYxwRyGk%2FQWOJrlfMWZjD%2Bp5egBjqyAUYFoGFBCbsX9c7nEJ5z9ZCPWxlzRKSHcfSG1CiEobQmJTFtiCruv%2FKx%2BeiMKnFH0%2BiPEogMxL3Xf8RHgjWM%2F3oWUsl1Xmw%2FJpVGaNAP%2BkjXGNNqBKh%2FdavDTzgTyUZwIS8k2VgbXI6bITamkytADvU090sJnQ8wsR6VZ9QXGpbrBni8yZdWJFVeuBYURwiwZFKmv0ItFGd0dg4CLVaw2OZLFM8qv6EINLs9c6AohDjC%2FFY%3D&Expires=1678235383 [following]\n",
            "--2023-03-06 12:29:43--  https://encode-public.s3.amazonaws.com/2016/10/21/c3715b23-624f-4b7e-b24d-64b0ef05ae7c/ENCFF693RXC.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF693RXC.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4255YSP7&Signature=ETkwLdnoBICtMvjrEOyWuv8TTQ8%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEFwaCXVzLXdlc3QtMiJGMEQCIFP35A%2BioOrW6IVGyQyHYyYnK8VaWSvGp7jUchETbjZoAiBiZMoW%2BFvH5J%2FIfFXDLZbTnVIwEF8iC%2BQCVFHmBTmu2CqyBQgVEAAaDDIyMDc0ODcxNDg2MyIMpkRbfQzFqfchY2RgKo8F0aK9MbO1M4n%2BBobVCDvcscKUl3f5Wa28elm3mwEKjduocXa%2FNTbNLlzZVzvpGU50cW1buIQ%2BltwRhOlXU0izF%2FeUsCqIsXPNH4J4o8lTeUBG%2F%2FeFKlNnBXC0eqtqPUOiW%2FOBG66AvMFJRV590qz2RHCzOkBhSUcMnMNyZr3oqKl3Hkez%2Fq3R8tJ69K3EiZ4mPj7x81yrXgUUC3k6lAi%2B8q2t3BGylTFSLBJnLCEiz%2BVWy13EjkLOeh75y2HOtV6jLKpWJsqy3kKfJBUcf9ZYEFFElkr0%2BGDgmvYdQNp6U41VC2tGcV4%2BND1qrxVn5o50VArBZjgjSHsROXPZLlov5M%2FS5akXggIsxkTkzDb0qbLkzxFRyHwr4reiOin5vHsQTElkKGupSy7ML7qMZ1NiPxE3eIFa74hj8H2%2B4XMQcjfqrmnPCX98FdForKNZe8GCxowodixuSztEciKO8bCzpasUuSSuGIUNBnFb9ZoSLZvfQZ2Vp9eeXam7%2FxP3mLmpVUPnDj26oUW0v1JYeFNtvfJXFOtqBGe0Os9gwfjHlvZoJUMyIvYs95IllG7HyiW%2FheIQRshad1ppLkKv9ZsRR2LP2TGOuLramQ1PWd9nYc2K1Gd%2BZx%2BslQpCCUjWWuCb3meCpPbEk%2B7wDUFKh30hGqH3p4zuBR95zoGBUs3ol4WB5Qrp7XlemMcP4nfy4Aaz0%2BCg10nJu9VAk7nt33GNHFsOb2eRTaHS%2F%2BEkXgfeRNOgOBFhc6Hx44Fq1WAJQfcME8Oluug1gUJwPKvQJr8kTZneH9Bw8iCgFKbKWMYigrAikG1t869Qg%2FaVQrKXSkYW03DOIIhWUIl%2By8zsIcql6ivYxwRyGk%2FQWOJrlfMWZjD%2Bp5egBjqyAUYFoGFBCbsX9c7nEJ5z9ZCPWxlzRKSHcfSG1CiEobQmJTFtiCruv%2FKx%2BeiMKnFH0%2BiPEogMxL3Xf8RHgjWM%2F3oWUsl1Xmw%2FJpVGaNAP%2BkjXGNNqBKh%2FdavDTzgTyUZwIS8k2VgbXI6bITamkytADvU090sJnQ8wsR6VZ9QXGpbrBni8yZdWJFVeuBYURwiwZFKmv0ItFGd0dg4CLVaw2OZLFM8qv6EINLs9c6AohDjC%2FFY%3D&Expires=1678235383\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.194.249, 52.218.252.91, 52.218.183.25, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.194.249|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2013652840 (1.9G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF693RXC.fastq.gz’\n",
            "\n",
            "ENCFF693RXC.fastq.g 100%[===================>]   1.88G  42.3MB/s    in 46s     \n",
            "\n",
            "2023-03-06 12:30:29 (42.0 MB/s) - ‘ENCFF693RXC.fastq.gz’ saved [2013652840/2013652840]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "# ChIP-seq на гистоновой метке\n",
        "!wget https://www.encodeproject.org/files/ENCFF090UYV/@@download/ENCFF090UYV.fastq.gz\n",
        "!wget https://www.encodeproject.org/files/ENCFF815LPM/@@download/ENCFF815LPM.fastq.gz\n",
        "# ChIP-seq контроль\n",
        "!wget https://www.encodeproject.org/files/ENCFF693RXC/@@download/ENCFF693RXC.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "v73fH7WiVDu9"
      },
      "outputs": [],
      "source": [
        "# распаковка ахрива\n",
        "!gzip -d ENCFF090UYV.fastq.gz ENCFF815LPM.fastq.gz ENCFF693RXC.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6sPcIefNHx-L"
      },
      "source": [
        "## FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5XM6jBcuGjMV",
        "outputId": "39828f1b-5606-4849-ed7d-76b4f1b67a5d"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Started analysis of ENCFF090UYV.fastq\n",
            "Approx 5% complete for ENCFF090UYV.fastq\n",
            "Approx 10% complete for ENCFF090UYV.fastq\n",
            "Approx 15% complete for ENCFF090UYV.fastq\n",
            "Approx 20% complete for ENCFF090UYV.fastq\n",
            "Approx 25% complete for ENCFF090UYV.fastq\n",
            "Approx 30% complete for ENCFF090UYV.fastq\n",
            "Approx 35% complete for ENCFF090UYV.fastq\n",
            "Approx 40% complete for ENCFF090UYV.fastq\n",
            "Approx 45% complete for ENCFF090UYV.fastq\n",
            "Approx 50% complete for ENCFF090UYV.fastq\n",
            "Approx 55% complete for ENCFF090UYV.fastq\n",
            "Approx 60% complete for ENCFF090UYV.fastq\n",
            "Approx 65% complete for ENCFF090UYV.fastq\n",
            "Approx 70% complete for ENCFF090UYV.fastq\n",
            "Approx 75% complete for ENCFF090UYV.fastq\n",
            "Approx 80% complete for ENCFF090UYV.fastq\n",
            "Approx 85% complete for ENCFF090UYV.fastq\n",
            "Approx 90% complete for ENCFF090UYV.fastq\n",
            "Approx 95% complete for ENCFF090UYV.fastq\n",
            "Analysis complete for ENCFF090UYV.fastq\n",
            "Started analysis of ENCFF815LPM.fastq\n",
            "Approx 5% complete for ENCFF815LPM.fastq\n",
            "Approx 10% complete for ENCFF815LPM.fastq\n",
            "Approx 15% complete for ENCFF815LPM.fastq\n",
            "Approx 20% complete for ENCFF815LPM.fastq\n",
            "Approx 25% complete for ENCFF815LPM.fastq\n",
            "Approx 30% complete for ENCFF815LPM.fastq\n",
            "Approx 35% complete for ENCFF815LPM.fastq\n",
            "Approx 40% complete for ENCFF815LPM.fastq\n",
            "Approx 45% complete for ENCFF815LPM.fastq\n",
            "Approx 50% complete for ENCFF815LPM.fastq\n",
            "Approx 55% complete for ENCFF815LPM.fastq\n",
            "Approx 60% complete for ENCFF815LPM.fastq\n",
            "Approx 65% complete for ENCFF815LPM.fastq\n",
            "Approx 70% complete for ENCFF815LPM.fastq\n",
            "Approx 75% complete for ENCFF815LPM.fastq\n",
            "Approx 80% complete for ENCFF815LPM.fastq\n",
            "Approx 85% complete for ENCFF815LPM.fastq\n",
            "Approx 90% complete for ENCFF815LPM.fastq\n",
            "Approx 95% complete for ENCFF815LPM.fastq\n",
            "Analysis complete for ENCFF815LPM.fastq\n",
            "Started analysis of ENCFF693RXC.fastq\n",
            "Approx 5% complete for ENCFF693RXC.fastq\n",
            "Approx 10% complete for ENCFF693RXC.fastq\n",
            "Approx 15% complete for ENCFF693RXC.fastq\n",
            "Approx 20% complete for ENCFF693RXC.fastq\n",
            "Approx 25% complete for ENCFF693RXC.fastq\n",
            "Approx 30% complete for ENCFF693RXC.fastq\n",
            "Approx 35% complete for ENCFF693RXC.fastq\n",
            "Approx 40% complete for ENCFF693RXC.fastq\n",
            "Approx 45% complete for ENCFF693RXC.fastq\n",
            "Approx 50% complete for ENCFF693RXC.fastq\n",
            "Approx 55% complete for ENCFF693RXC.fastq\n",
            "Approx 60% complete for ENCFF693RXC.fastq\n",
            "Approx 65% complete for ENCFF693RXC.fastq\n",
            "Approx 70% complete for ENCFF693RXC.fastq\n",
            "Approx 75% complete for ENCFF693RXC.fastq\n",
            "Approx 80% complete for ENCFF693RXC.fastq\n",
            "Approx 85% complete for ENCFF693RXC.fastq\n",
            "Approx 90% complete for ENCFF693RXC.fastq\n",
            "Approx 95% complete for ENCFF693RXC.fastq\n",
            "Analysis complete for ENCFF693RXC.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc /content/ENCFF090UYV.fastq\n",
        "!./FastQC/fastqc /content/ENCFF815LPM.fastq\n",
        "!./FastQC/fastqc /content/ENCFF693RXC.fastq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "BGz7oH4W7vfE"
      },
      "outputs": [],
      "source": [
        "!cp *.html /content/drive/MyDrive/hse23_hw2/"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W9QHWBLlpMjQ"
      },
      "source": [
        "## Выравнивание на хромосому"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "N4JIGZOsGiY2",
        "outputId": "30cc8d2d-cea1-4379-ecf4-70ee997f8556"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2023-03-06 12:51:56--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 29295890 (28M) [application/x-gzip]\n",
            "Saving to: ‘chr14.fa.gz’\n",
            "\n",
            "chr14.fa.gz         100%[===================>]  27.94M  30.0MB/s    in 0.9s    \n",
            "\n",
            "2023-03-06 12:51:58 (30.0 MB/s) - ‘chr14.fa.gz’ saved [29295890/29295890]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
        "!gzip -d chr14.fa.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bhraQ6NdGiPN",
        "outputId": "934f7942-b7a7-4f25-f1f9-3fe913ba0617"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Settings:\n",
            "  Output files: \"chromosome_index.*.bt2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Max bucket size: default\n",
            "  Max bucket size, sqrt multiplier: default\n",
            "  Max bucket size, len divisor: 4\n",
            "  Difference-cover sample period: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  chr14.fa\n",
            "Building a SMALL index\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:00\n",
            "bmax according to bmaxDivN setting: 22642037\n",
            "Using parameters --bmax 16981528 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 16981528 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 1, merged 5; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 1.1321e+07 (target: 16981527)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 8\n",
            "  Reserving size (16981528) for bucket 1\n",
            "  Calculating Z arrays for bucket 1\n",
            "  Entering block accumulator loop for bucket 1:\n",
            "  bucket 1: 10%\n",
            "  bucket 1: 20%\n",
            "  bucket 1: 30%\n",
            "  bucket 1: 40%\n",
            "  bucket 1: 50%\n",
            "  bucket 1: 60%\n",
            "  bucket 1: 70%\n",
            "  bucket 1: 80%\n",
            "  bucket 1: 90%\n",
            "  bucket 1: 100%\n",
            "  Sorting block of length 10995868 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:04\n",
            "Returning block of 10995869 for bucket 1\n",
            "Getting block 2 of 8\n",
            "  Reserving size (16981528) for bucket 2\n",
            "  Calculating Z arrays for bucket 2\n",
            "  Entering block accumulator loop for bucket 2:\n",
            "  bucket 2: 10%\n",
            "  bucket 2: 20%\n",
            "  bucket 2: 30%\n",
            "  bucket 2: 40%\n",
            "  bucket 2: 50%\n",
            "  bucket 2: 60%\n",
            "  bucket 2: 70%\n",
            "  bucket 2: 80%\n",
            "  bucket 2: 90%\n",
            "  bucket 2: 100%\n",
            "  Sorting block of length 10732981 for bucket 2\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:03\n",
            "Returning block of 10732982 for bucket 2\n",
            "Getting block 3 of 8\n",
            "  Reserving size (16981528) for bucket 3\n",
            "  Calculating Z arrays for bucket 3\n",
            "  Entering block accumulator loop for bucket 3:\n",
            "  bucket 3: 10%\n",
            "  bucket 3: 20%\n",
            "  bucket 3: 30%\n",
            "  bucket 3: 40%\n",
            "  bucket 3: 50%\n",
            "  bucket 3: 60%\n",
            "  bucket 3: 70%\n",
            "  bucket 3: 80%\n",
            "  bucket 3: 90%\n",
            "  bucket 3: 100%\n",
            "  Sorting block of length 14209021 for bucket 3\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:05\n",
            "Returning block of 14209022 for bucket 3\n",
            "Getting block 4 of 8\n",
            "  Reserving size (16981528) for bucket 4\n",
            "  Calculating Z arrays for bucket 4\n",
            "  Entering block accumulator loop for bucket 4:\n",
            "  bucket 4: 10%\n",
            "  bucket 4: 20%\n",
            "  bucket 4: 30%\n",
            "  bucket 4: 40%\n",
            "  bucket 4: 50%\n",
            "  bucket 4: 60%\n",
            "  bucket 4: 70%\n",
            "  bucket 4: 80%\n",
            "  bucket 4: 90%\n",
            "  bucket 4: 100%\n",
            "  Sorting block of length 16396496 for bucket 4\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:04\n",
            "Returning block of 16396497 for bucket 4\n",
            "Getting block 5 of 8\n",
            "  Reserving size (16981528) for bucket 5\n",
            "  Calculating Z arrays for bucket 5\n",
            "  Entering block accumulator loop for bucket 5:\n",
            "  bucket 5: 10%\n",
            "  bucket 5: 20%\n",
            "  bucket 5: 30%\n",
            "  bucket 5: 40%\n",
            "  bucket 5: 50%\n",
            "  bucket 5: 60%\n",
            "  bucket 5: 70%\n",
            "  bucket 5: 80%\n",
            "  bucket 5: 90%\n",
            "  bucket 5: 100%\n",
            "  Sorting block of length 6586600 for bucket 5\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:02\n",
            "Returning block of 6586601 for bucket 5\n",
            "Getting block 6 of 8\n",
            "  Reserving size (16981528) for bucket 6\n",
            "  Calculating Z arrays for bucket 6\n",
            "  Entering block accumulator loop for bucket 6:\n",
            "  bucket 6: 10%\n",
            "  bucket 6: 20%\n",
            "  bucket 6: 30%\n",
            "  bucket 6: 40%\n",
            "  bucket 6: 50%\n",
            "  bucket 6: 60%\n",
            "  bucket 6: 70%\n",
            "  bucket 6: 80%\n",
            "  bucket 6: 90%\n",
            "  bucket 6: 100%\n",
            "  Sorting block of length 14286880 for bucket 6\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:04\n",
            "Returning block of 14286881 for bucket 6\n",
            "Getting block 7 of 8\n",
            "  Reserving size (16981528) for bucket 7\n",
            "  Calculating Z arrays for bucket 7\n",
            "  Entering block accumulator loop for bucket 7:\n",
            "  bucket 7: 10%\n",
            "  bucket 7: 20%\n",
            "  bucket 7: 30%\n",
            "  bucket 7: 40%\n",
            "  bucket 7: 50%\n",
            "  bucket 7: 60%\n",
            "  bucket 7: 70%\n",
            "  bucket 7: 80%\n",
            "  bucket 7: 90%\n",
            "  bucket 7: 100%\n",
            "  Sorting block of length 16145458 for bucket 7\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:04\n",
            "Returning block of 16145459 for bucket 7\n",
            "Getting block 8 of 8\n",
            "  Reserving size (16981528) for bucket 8\n",
            "  Calculating Z arrays for bucket 8\n",
            "  Entering block accumulator loop for bucket 8:\n",
            "  bucket 8: 10%\n",
            "  bucket 8: 20%\n",
            "  bucket 8: 30%\n",
            "  bucket 8: 40%\n",
            "  bucket 8: 50%\n",
            "  bucket 8: 60%\n",
            "  bucket 8: 70%\n",
            "  bucket 8: 80%\n",
            "  bucket 8: 90%\n",
            "  bucket 8: 100%\n",
            "  Sorting block of length 1214838 for bucket 8\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:00\n",
            "Returning block of 1214839 for bucket 8\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 26673415\n",
            "fchr[G]: 45097173\n",
            "fchr[T]: 63656206\n",
            "fchr[$]: 90568149\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 34384187 bytes to primary EBWT file: chromosome_index.1.bt2.tmp\n",
            "Wrote 22642044 bytes to secondary EBWT file: chromosome_index.2.bt2.tmp\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 90568149\n",
            "    bwtLen: 90568150\n",
            "    sz: 22642038\n",
            "    bwtSz: 22642038\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5660510\n",
            "    offsSz: 22642040\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 471710\n",
            "    numLines: 471710\n",
            "    ebwtTotLen: 30189440\n",
            "    ebwtTotSz: 30189440\n",
            "    color: 0\n",
            "    reverse: 0\n",
            "Total time for call to driver() for forward index: 00:01:07\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:00\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "  Time to reverse reference sequence: 00:00:00\n",
            "bmax according to bmaxDivN setting: 22642037\n",
            "Using parameters --bmax 16981528 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 16981528 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 1, merged 7; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 1, merged 0; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 1.29383e+07 (target: 16981527)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 7\n",
            "  Reserving size (16981528) for bucket 1\n",
            "  Calculating Z arrays for bucket 1\n",
            "  Entering block accumulator loop for bucket 1:\n",
            "  bucket 1: 10%\n",
            "  bucket 1: 20%\n",
            "  bucket 1: 30%\n",
            "  bucket 1: 40%\n",
            "  bucket 1: 50%\n",
            "  bucket 1: 60%\n",
            "  bucket 1: 70%\n",
            "  bucket 1: 80%\n",
            "  bucket 1: 90%\n",
            "  bucket 1: 100%\n",
            "  Sorting block of length 12277527 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:05\n",
            "Returning block of 12277528 for bucket 1\n",
            "Getting block 2 of 7\n",
            "  Reserving size (16981528) for bucket 2\n",
            "  Calculating Z arrays for bucket 2\n",
            "  Entering block accumulator loop for bucket 2:\n",
            "  bucket 2: 10%\n",
            "  bucket 2: 20%\n",
            "  bucket 2: 30%\n",
            "  bucket 2: 40%\n",
            "  bucket 2: 50%\n",
            "  bucket 2: 60%\n",
            "  bucket 2: 70%\n",
            "  bucket 2: 80%\n",
            "  bucket 2: 90%\n",
            "  bucket 2: 100%\n",
            "  Sorting block of length 16458020 for bucket 2\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:04\n",
            "Returning block of 16458021 for bucket 2\n",
            "Getting block 3 of 7\n",
            "  Reserving size (16981528) for bucket 3\n",
            "  Calculating Z arrays for bucket 3\n",
            "  Entering block accumulator loop for bucket 3:\n",
            "  bucket 3: 10%\n",
            "  bucket 3: 20%\n",
            "  bucket 3: 30%\n",
            "  bucket 3: 40%\n",
            "  bucket 3: 50%\n",
            "  bucket 3: 60%\n",
            "  bucket 3: 70%\n",
            "  bucket 3: 80%\n",
            "  bucket 3: 90%\n",
            "  bucket 3: 100%\n",
            "  Sorting block of length 15884098 for bucket 3\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:06\n",
            "Returning block of 15884099 for bucket 3\n",
            "Getting block 4 of 7\n",
            "  Reserving size (16981528) for bucket 4\n",
            "  Calculating Z arrays for bucket 4\n",
            "  Entering block accumulator loop for bucket 4:\n",
            "  bucket 4: 10%\n",
            "  bucket 4: 20%\n",
            "  bucket 4: 30%\n",
            "  bucket 4: 40%\n",
            "  bucket 4: 50%\n",
            "  bucket 4: 60%\n",
            "  bucket 4: 70%\n",
            "  bucket 4: 80%\n",
            "  bucket 4: 90%\n",
            "  bucket 4: 100%\n",
            "  Sorting block of length 10967214 for bucket 4\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:02\n",
            "Returning block of 10967215 for bucket 4\n",
            "Getting block 5 of 7\n",
            "  Reserving size (16981528) for bucket 5\n",
            "  Calculating Z arrays for bucket 5\n",
            "  Entering block accumulator loop for bucket 5:\n",
            "  bucket 5: 10%\n",
            "  bucket 5: 20%\n",
            "  bucket 5: 30%\n",
            "  bucket 5: 40%\n",
            "  bucket 5: 50%\n",
            "  bucket 5: 60%\n",
            "  bucket 5: 70%\n",
            "  bucket 5: 80%\n",
            "  bucket 5: 90%\n",
            "  bucket 5: 100%\n",
            "  Sorting block of length 10048387 for bucket 5\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:03\n",
            "Returning block of 10048388 for bucket 5\n",
            "Getting block 6 of 7\n",
            "  Reserving size (16981528) for bucket 6\n",
            "  Calculating Z arrays for bucket 6\n",
            "  Entering block accumulator loop for bucket 6:\n",
            "  bucket 6: 10%\n",
            "  bucket 6: 20%\n",
            "  bucket 6: 30%\n",
            "  bucket 6: 40%\n",
            "  bucket 6: 50%\n",
            "  bucket 6: 60%\n",
            "  bucket 6: 70%\n",
            "  bucket 6: 80%\n",
            "  bucket 6: 90%\n",
            "  bucket 6: 100%\n",
            "  Sorting block of length 11270887 for bucket 6\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:04\n",
            "Returning block of 11270888 for bucket 6\n",
            "Getting block 7 of 7\n",
            "  Reserving size (16981528) for bucket 7\n",
            "  Calculating Z arrays for bucket 7\n",
            "  Entering block accumulator loop for bucket 7:\n",
            "  bucket 7: 10%\n",
            "  bucket 7: 20%\n",
            "  bucket 7: 30%\n",
            "  bucket 7: 40%\n",
            "  bucket 7: 50%\n",
            "  bucket 7: 60%\n",
            "  bucket 7: 70%\n",
            "  bucket 7: 80%\n",
            "  bucket 7: 90%\n",
            "  bucket 7: 100%\n",
            "  Sorting block of length 13662010 for bucket 7\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:03\n",
            "Returning block of 13662011 for bucket 7\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 26673415\n",
            "fchr[G]: 45097173\n",
            "fchr[T]: 63656206\n",
            "fchr[$]: 90568149\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 34384187 bytes to primary EBWT file: chromosome_index.rev.1.bt2.tmp\n",
            "Wrote 22642044 bytes to secondary EBWT file: chromosome_index.rev.2.bt2.tmp\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 90568149\n",
            "    bwtLen: 90568150\n",
            "    sz: 22642038\n",
            "    bwtSz: 22642038\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5660510\n",
            "    offsSz: 22642040\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 471710\n",
            "    numLines: 471710\n",
            "    ebwtTotLen: 30189440\n",
            "    ebwtTotSz: 30189440\n",
            "    color: 0\n",
            "    reverse: 1\n",
            "Total time for backward call to driver() for mirror index: 00:01:08\n",
            "Renaming chromosome_index.3.bt2.tmp to chromosome_index.3.bt2\n",
            "Renaming chromosome_index.4.bt2.tmp to chromosome_index.4.bt2\n",
            "Renaming chromosome_index.1.bt2.tmp to chromosome_index.1.bt2\n",
            "Renaming chromosome_index.2.bt2.tmp to chromosome_index.2.bt2\n",
            "Renaming chromosome_index.rev.1.bt2.tmp to chromosome_index.rev.1.bt2\n",
            "Renaming chromosome_index.rev.2.bt2.tmp to chromosome_index.rev.2.bt2\n",
            "CPU times: user 1.3 s, sys: 138 ms, total: 1.44 s\n",
            "Wall time: 2min 15s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2-build chr14.fa chromosome_index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "E_usI8XUrQnR"
      },
      "outputs": [],
      "source": [
        "!mkdir bowtie2_res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3t80UATHqFe7",
        "outputId": "ce70998d-fe3f-45b7-a473-04739e7540ce"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "43763763 reads; of these:\n",
            "  43763763 (100.00%) were unpaired; of these:\n",
            "    39692520 (90.70%) aligned 0 times\n",
            "    1144629 (2.62%) aligned exactly 1 time\n",
            "    2926614 (6.69%) aligned >1 times\n",
            "9.30% overall alignment rate\n",
            "CPU times: user 8.46 s, sys: 784 ms, total: 9.24 s\n",
            "Wall time: 17min 10s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF090UYV.fastq \\\n",
        "          -S bowtie2_res/res_UYV.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "so1jTdGHoEir",
        "outputId": "656c803c-f9c1-4e97-9f6a-0c1d9dd93619"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "33979765 reads; of these:\n",
            "  33979765 (100.00%) were unpaired; of these:\n",
            "    29585894 (87.07%) aligned 0 times\n",
            "    1201637 (3.54%) aligned exactly 1 time\n",
            "    3192234 (9.39%) aligned >1 times\n",
            "12.93% overall alignment rate\n",
            "CPU times: user 7.76 s, sys: 719 ms, total: 8.48 s\n",
            "Wall time: 15min 11s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF815LPM.fastq \\\n",
        "          -S bowtie2_res/res_LPM.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8Hn8n84v-EQY",
        "outputId": "14489232-b40e-420f-c9d8-e3447e52cbe6"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "42675605 reads; of these:\n",
            "  42675605 (100.00%) were unpaired; of these:\n",
            "    35483816 (83.15%) aligned 0 times\n",
            "    1790757 (4.20%) aligned exactly 1 time\n",
            "    5401032 (12.66%) aligned >1 times\n",
            "16.85% overall alignment rate\n",
            "CPU times: user 12 s, sys: 1.2 s, total: 13.2 s\n",
            "Wall time: 23min 21s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF693RXC.fastq \\\n",
        "          -S bowtie2_res/res_RXC.sam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ysoDrnSCrei0"
      },
      "source": [
        "Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?\n",
        "\n",
        "Имеет смысл для дальнейшего анализа отобрать уникально картированные риды."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XZtQrPZMmgnC"
      },
      "source": [
        "## Peak calling"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "uX1ps4VDp_Ae"
      },
      "outputs": [],
      "source": [
        "!mkdir macs2"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EJQOCcv2nJW_",
        "outputId": "1c59c2bf-83e5-48d5-b39a-85d332cf45fe"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "INFO  @ Mon, 06 Mar 2023 15:06:18: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_UYV.sam -c bowtie2_res/res_RXC.sam -f SAM --outdir macs2 -n ENCFF090UYV\n",
            "# ARGUMENTS LIST:\n",
            "# name = ENCFF090UYV\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_UYV.sam']\n",
            "# control file = ['bowtie2_res/res_RXC.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Mon, 06 Mar 2023 15:06:18: #1 read tag files... \n",
            "INFO  @ Mon, 06 Mar 2023 15:06:18: #1 read treatment tags... \n",
            "INFO  @ Mon, 06 Mar 2023 15:06:19:  1000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:06:28:  7000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:06:33:  12000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:06:38:  17000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:06:40:  18000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:04: #1.2 read input tags... \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:17:  13000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:21:  16000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:31:  25000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:36:  29000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:40:  32000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:42:  34000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1 tag size is determined as 36 bps \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1 tag size = 36.0 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1  total tags in treatment: 4071243 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1  tags after filtering in treatment: 3319183 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1  Redundant rate of treatment: 0.18 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1  total tags in control: 7191789 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1  tags after filtering in control: 5821248 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1  Redundant rate of control: 0.19 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #1 finished! \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #2 Build Peak Model... \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:53: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: #2 number of paired peaks: 33424 \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: start model_add_line... \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: start X-correlation... \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: end of X-cor \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: #2 finished! \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: #2 predicted fragment length is 36 bps \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: #2 alternative fragment length(s) may be 36,491 bps \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: #2.2 Generate R script for model : macs2/ENCFF090UYV_model.r \n",
            "WARNING @ Mon, 06 Mar 2023 15:07:54: #2 Since the d (36) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Mon, 06 Mar 2023 15:07:54: #2 You may need to consider one of the other alternative d(s): 36,491 \n",
            "WARNING @ Mon, 06 Mar 2023 15:07:54: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: #3 Call peaks... \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Mon, 06 Mar 2023 15:07:54: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Mon, 06 Mar 2023 15:08:17: #3 Call peaks for each chromosome... \n",
            "INFO  @ Mon, 06 Mar 2023 15:08:31: #4 Write output xls file... macs2/ENCFF090UYV_peaks.xls \n",
            "INFO  @ Mon, 06 Mar 2023 15:08:31: #4 Write broad peak in broadPeak format file... macs2/ENCFF090UYV_peaks.broadPeak \n",
            "INFO  @ Mon, 06 Mar 2023 15:08:31: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF090UYV_peaks.gappedPeak \n",
            "INFO  @ Mon, 06 Mar 2023 15:08:31: Done! \n"
          ]
        }
      ],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_UYV.sam \\\n",
        "    -c bowtie2_res/res_RXC.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2 \\\n",
        "\t  -n ENCFF090UYV"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jQnB-d9wZwPt",
        "outputId": "0a4ea5d2-c8c7-4c6e-d294-d17e4955b20b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "INFO  @ Mon, 06 Mar 2023 15:08:31: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_LPM.sam -c bowtie2_res/res_RXC.sam -f SAM --outdir macs2 -n ENCFF815LPM\n",
            "# ARGUMENTS LIST:\n",
            "# name = ENCFF815LPM\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_LPM.sam']\n",
            "# control file = ['bowtie2_res/res_RXC.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Mon, 06 Mar 2023 15:08:31: #1 read tag files... \n",
            "INFO  @ Mon, 06 Mar 2023 15:08:31: #1 read treatment tags... \n",
            "INFO  @ Mon, 06 Mar 2023 15:08:34:  3000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:08:44:  11000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:09:02:  20000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:09:15:  25000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:09:25: #1.2 read input tags... \n",
            "INFO  @ Mon, 06 Mar 2023 15:09:40:  13000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:09:43:  16000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:09:53:  25000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:09:59:  29000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:02:  32000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:04:  34000000 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1 tag size is determined as 36 bps \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1 tag size = 36.0 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1  total tags in treatment: 4393871 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1  tags after filtering in treatment: 3637614 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1  Redundant rate of treatment: 0.17 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1  total tags in control: 7191789 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1  tags after filtering in control: 5821248 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1  Redundant rate of control: 0.19 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #1 finished! \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #2 Build Peak Model... \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:15: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: #2 number of paired peaks: 33897 \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: start model_add_line... \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: start X-correlation... \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: end of X-cor \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: #2 finished! \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: #2 predicted fragment length is 35 bps \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: #2 alternative fragment length(s) may be 35 bps \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: #2.2 Generate R script for model : macs2/ENCFF815LPM_model.r \n",
            "WARNING @ Mon, 06 Mar 2023 15:10:16: #2 Since the d (35) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Mon, 06 Mar 2023 15:10:16: #2 You may need to consider one of the other alternative d(s): 35 \n",
            "WARNING @ Mon, 06 Mar 2023 15:10:16: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: #3 Call peaks... \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:16: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:39: #3 Call peaks for each chromosome... \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:54: #4 Write output xls file... macs2/ENCFF815LPM_peaks.xls \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:54: #4 Write broad peak in broadPeak format file... macs2/ENCFF815LPM_peaks.broadPeak \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:54: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF815LPM_peaks.gappedPeak \n",
            "INFO  @ Mon, 06 Mar 2023 15:10:54: Done! \n"
          ]
        }
      ],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_LPM.sam \\\n",
        "    -c bowtie2_res/res_RXC.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2 \\\n",
        "    -n ENCFF815LPM"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HrNwxs7ArPz7",
        "outputId": "15febd81-80cf-4242-ee1a-179e286f7fa1"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2023-03-06 15:11:24--  https://www.encodeproject.org/files/ENCFF946FXB/@@download/ENCFF946FXB.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/01/31/d5cbc20c-74e8-40d5-964c-39cbaebaed1a/ENCFF946FXB.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF946FXB.bed.gz&AWSAccessKeyId=ASIATGZNGCNXWI7F4IIY&Signature=bvjxwPpIOpnx8vZA1Dn99LZPMSE%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEF4aCXVzLXdlc3QtMiJHMEUCIQDSKSjMjq73MPOOrC1TJ1z4cds33mCvHTrjsmOo8HXkKAIgCf4bo51L3zElEjtSntE%2Fgf6gcszgDtJ2htjsy0bz%2FZ8qsgUIFxAAGgwyMjA3NDg3MTQ4NjMiDABy3lnzI7EAEwmRdSqPBW%2F6VOSLZbj26bF5h1hGBTHitdhUVveQ6qSQo9IRfJnNBdEkexhYUcJIdFaw4JkkGH5DAnfdQDb7heVUd%2FyvYSdrB4TJbINXSLTiZGJAfE9HoCfO2Sc%2BsSe3fLp6zAdr9T6yJvo6rewmvMazzUWpBBRDF%2BExXigvaA%2F3NpAQ8Eb%2B4YxZCiMKR%2FhrJhmFWR2z5qngySBYYu%2FoFpQg4Lh5wLBzFfkpkL%2FGYndnp4SqY2em%2FripUJ4iW1rxJVGYgcjMUOgETt%2FZF65hHuCTteotUj%2Fn81HpGZsQLcZG3%2FtfDRlN1g%2FSh1R%2B%2FbdP3T5H%2Blbip0nLvSNbtQdDsJuHLN96QlTcOZ6%2BaTPNY8b1GucXOW32knwdSSMkylGcDo6H7raxxBN8SqNl47dJ2TmVVijNL0byNKsHi5x3Kv%2Brl1kvWKDyoPSYlTIEfqKaY3fJFH7UEU1sngw9rIL4SWpkFnQUDLE9LF2WdGMOmEojkSx0gBWtNGPvdX1ndKUz10RkIX3vacPUS5iX74tnI5gpoMA%2BOFqZGCGAdFeEmuNFZ4gG48zKeLyPCNbi%2B10m%2B7sIt5VyrktW7yQH9mFUuIbaSGIDMcnc0SUPejku46YSbCzR6ihV34Punq8OdkQ%2BLJUf1H5QiPdXjMy7BhTqeF0moOBBCTsWc7%2Bu5Fx0W%2F2Ei0bcdhSKP8zDROswri6Mgj1YsL4zDz5Y1rgg971o%2Bx6Ao%2BifuyfU6VwdRKF%2FgjGqud6C%2FhNzJbxU9Ynho0gAbXasoiMyWOAdZ%2F1JgD3knXTMhGsDfezq7s1kgPsoNGAHkyWEjJslxntTdRVj%2FVBRRU5fdZ249%2FCvb0FNZ13WJWuqvkqFOiWcqa1zoOPq8q4tu0RuGMww6eeXoAY6sQEtyetoFnkpzCFKd7NibIwV%2F%2FOtPm6XOUlmgnk2xk2qWGmJB0NRVnBOCwFaGYnn7EkGe87ix%2FFa%2B%2F1hrbxBOpzdAIooZYUvs%2F29q4s1KUAwry9VmDW48AgrifGQt6AEQeM9oYRp2eP%2BhnWNXsGMw6yuMCC%2BLZ0WSSa%2BY9oi3x%2BAsZpOOxF%2Bwt3UjBnYwM3ykDNsotgMNUp4vSg%2FTyRZmDiUYevzRNSubWECHvH9nseWLQE%3D&Expires=1678245084 [following]\n",
            "--2023-03-06 15:11:24--  https://encode-public.s3.amazonaws.com/2017/01/31/d5cbc20c-74e8-40d5-964c-39cbaebaed1a/ENCFF946FXB.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF946FXB.bed.gz&AWSAccessKeyId=ASIATGZNGCNXWI7F4IIY&Signature=bvjxwPpIOpnx8vZA1Dn99LZPMSE%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEF4aCXVzLXdlc3QtMiJHMEUCIQDSKSjMjq73MPOOrC1TJ1z4cds33mCvHTrjsmOo8HXkKAIgCf4bo51L3zElEjtSntE%2Fgf6gcszgDtJ2htjsy0bz%2FZ8qsgUIFxAAGgwyMjA3NDg3MTQ4NjMiDABy3lnzI7EAEwmRdSqPBW%2F6VOSLZbj26bF5h1hGBTHitdhUVveQ6qSQo9IRfJnNBdEkexhYUcJIdFaw4JkkGH5DAnfdQDb7heVUd%2FyvYSdrB4TJbINXSLTiZGJAfE9HoCfO2Sc%2BsSe3fLp6zAdr9T6yJvo6rewmvMazzUWpBBRDF%2BExXigvaA%2F3NpAQ8Eb%2B4YxZCiMKR%2FhrJhmFWR2z5qngySBYYu%2FoFpQg4Lh5wLBzFfkpkL%2FGYndnp4SqY2em%2FripUJ4iW1rxJVGYgcjMUOgETt%2FZF65hHuCTteotUj%2Fn81HpGZsQLcZG3%2FtfDRlN1g%2FSh1R%2B%2FbdP3T5H%2Blbip0nLvSNbtQdDsJuHLN96QlTcOZ6%2BaTPNY8b1GucXOW32knwdSSMkylGcDo6H7raxxBN8SqNl47dJ2TmVVijNL0byNKsHi5x3Kv%2Brl1kvWKDyoPSYlTIEfqKaY3fJFH7UEU1sngw9rIL4SWpkFnQUDLE9LF2WdGMOmEojkSx0gBWtNGPvdX1ndKUz10RkIX3vacPUS5iX74tnI5gpoMA%2BOFqZGCGAdFeEmuNFZ4gG48zKeLyPCNbi%2B10m%2B7sIt5VyrktW7yQH9mFUuIbaSGIDMcnc0SUPejku46YSbCzR6ihV34Punq8OdkQ%2BLJUf1H5QiPdXjMy7BhTqeF0moOBBCTsWc7%2Bu5Fx0W%2F2Ei0bcdhSKP8zDROswri6Mgj1YsL4zDz5Y1rgg971o%2Bx6Ao%2BifuyfU6VwdRKF%2FgjGqud6C%2FhNzJbxU9Ynho0gAbXasoiMyWOAdZ%2F1JgD3knXTMhGsDfezq7s1kgPsoNGAHkyWEjJslxntTdRVj%2FVBRRU5fdZ249%2FCvb0FNZ13WJWuqvkqFOiWcqa1zoOPq8q4tu0RuGMww6eeXoAY6sQEtyetoFnkpzCFKd7NibIwV%2F%2FOtPm6XOUlmgnk2xk2qWGmJB0NRVnBOCwFaGYnn7EkGe87ix%2FFa%2B%2F1hrbxBOpzdAIooZYUvs%2F29q4s1KUAwry9VmDW48AgrifGQt6AEQeM9oYRp2eP%2BhnWNXsGMw6yuMCC%2BLZ0WSSa%2BY9oi3x%2BAsZpOOxF%2Bwt3UjBnYwM3ykDNsotgMNUp4vSg%2FTyRZmDiUYevzRNSubWECHvH9nseWLQE%3D&Expires=1678245084\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.138.113, 52.92.147.17, 52.92.243.105, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.138.113|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1040718 (1016K) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF946FXB.bed.gz’\n",
            "\n",
            "ENCFF946FXB.bed.gz  100%[===================>]   1016K  3.02MB/s    in 0.3s    \n",
            "\n",
            "2023-03-06 15:11:25 (3.02 MB/s) - ‘ENCFF946FXB.bed.gz’ saved [1040718/1040718]\n",
            "\n",
            "gzip: ENCFF367KIF.bed.gz: No such file or directory\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF946FXB/@@download/ENCFF946FXB.bed.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "mxz7OObNe7Sl"
      },
      "outputs": [],
      "source": [
        "!gzip -d ENCFF946FXB.bed.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MYEGEbQjvEwY"
      },
      "source": [
        "## Сравнение результатов\n",
        "\n",
        "Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.\n",
        "\n",
        "Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_1EAygSCvc8a",
        "outputId": "4419681b-7bec-48df-954c-aff38fdb8128"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn11.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2/ENCFF090UYV_peaks.broadPeak ENCFF946FXB.bed --filenames --output venn_results/venn11.jpg"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7qqn3n2cRdkt",
        "outputId": "1cc95bde-01f4-462d-b9e4-328432cff2fd"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn12.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i ENCFF946FXB.bed macs2/ENCFF090UYV_peaks.broadPeak --filenames --output venn_results/venn12.jpg"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "odjUzVjpfWWI",
        "outputId": "d04a0c53-8ec3-4190-f637-dd241adb9f9b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn21.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2/ENCFF815LPM_peaks.broadPeak ENCFF946FXB.bed --filenames --output venn_results/venn21.jpg"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IlXJFQJBfXr_",
        "outputId": "26ba26ab-8946-4464-880e-5cc609b175e4"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn22.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i ENCFF946FXB.bed macs2/ENCFF815LPM_peaks.broadPeak --filenames --output venn_results/venn22.jpg"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}